realm/realm-library/src/main/cpp/object-store/src/shared_realm.hpp

```diff
@@ -118,8 +118,12 @@ namespace realm {
         void cancel_transaction();
         bool is_in_transaction() const noexcept;
         bool is_in_read_transaction() const { return !!m_group; }
+        SharedGroup::VersionID get_version_of_current_transaction() const {
+            return m_shared_group->get_version_of_current_transaction();
+        }
 
-        bool refresh();
+        // FIXME: The param should not be needed.
+        bool refresh(SharedGroup::VersionID version = {});
```

Comment:
@bdash This might not be suitable. All java needs is advance to the latest/a specific version. And refresh to a specific version doesn't seem to be a generic requirement for others. So maybe I should add a new API somewhere? Something like `SharedRealm::advance(Version)` .